### PR TITLE
aws s3 integration

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
AWS S3 pre-canned integration

### Issues Resolved
- https://github.com/opensearch-project/dashboards-observability/issues/684

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
dashboard

File: server/adaptors/integrations/__data__/repository/aws_s3/schemas/aws_s3-1.0.0.mapping.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                  "type": "text",
+                  "fields": {
+                    "keyword": {
+                      "type": "keyword",
+                      "ignore_above": 256
+                    }
+                  }
+                },
+                "http_status": {
+                  "type": "keyword"
```
Comment (Swiddis) [inline server/adaptors/integrations/__data__/repository/aws_s3/schemas/aws_s3-1.0.0.mapping.json:88]: Keyword in schema but int in data

File: docker-compose.yml
Comment (Swiddis) [inline docker-compose.yml:1]: Modifying other external files within one integration PR again

**Review by Swiddis — COMMENTED** [review]
Confirmed working in local pull, minor comments on type checking and adding too many files

File: server/adaptors/integrations/__data__/repository/aws_s3/data/sample.json [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+      },
+      "region": "ap-southeast-2",
+      "resource_id": "s3-centralizedlogging-webconsoleuis3bucket22191f5e-t9bxwwr3d7k",
+      "platform": "aws_s3"
+    },
+    "aws": {
+      "s3": {
+        "bucket_owner": "078d85edf7268fb4814b1b4fc9f4c64adfde218b6b489a38ecf1b269f14f3c7a",
+        "bucket": "centralizedlogging-webconsoleuis3bucket22191f5e-t9bxwwr3d7k",
+        "time": "31/Jan/2023:09:25:20 +0000",
```
Comment (Swiddis) [inline server/adaptors/integrations/__data__/repository/aws_s3/data/sample.json:30]: `aws.s3.time` isn't present in any of the schemas

File: server/adaptors/integrations/__data__/repository/aws_s3/schemas/aws_s3-1.0.0.mapping.json (hunk 1 above)
Comment (YANG-DB) [reply server/adaptors/integrations/__data__/repository/aws_s3/schemas/aws_s3-1.0.0.mapping.json:88]: will update the sample.json

File: server/adaptors/integrations/__data__/repository/aws_s3/data/sample.json (hunk 2 above)
Comment (YANG-DB) [reply server/adaptors/integrations/__data__/repository/aws_s3/data/sample.json:30]: will update this

**Review by Swiddis — APPROVED** [review]

**Review by anirudha — APPROVED** [review]

**Review by derek-ho — DISMISSED** [review]
Can we check we are compliant with OpenSearch data format on object store. Also looks like merge conflict

**Pull request merged**